lib/elasticity/index_config.rb

```diff
@@ -42,9 +53,13 @@ def strategy
     private
 
     def validate!
-      ATTRS.each do |attr|
+      validatable_attributes.each do |attr|
         raise "#{attr} is not set" if public_send(attr).nil?
       end
     end
+
+    def validatable_attributes
+      ATTRS - [:mapping, :subclasses]
+    end
```

Comment:
Why not make this a constant too? No need to create this array over and over again.